src/index.js

```diff
@@ -211,23 +200,25 @@ class Router extends Component {
 	}
 
 	getMatchingChildren(children, url, invoke) {
-		return children.slice().sort(pathRankSort).filter( ({ attributes }) => {
-			let path = attributes.path,
-				matches = exec(url, path, attributes);
-			if (matches) {
-				if (invoke!==false) {
-					attributes.url = url;
-					attributes.matches = matches;
-					// copy matches onto props
-					for (let i in matches) {
-						if (matches.hasOwnProperty(i)) {
-							attributes[i] = matches[i];
-						}
+		return children
+			.map((child, index) => ({ child, index, rank: rankChild(child) }))
+			.sort((a, b) => (
+				(a.rank < b.rank) ? 1 :
+				(a.rank > b.rank) ? -1 :
+				(a.index - b.index)
+			))
+			.map(({ child }) => child)
+			.filter(({ attributes=EMPTY }) => {
+				let path = attributes.path,
+					matches = exec(url, path, attributes);
+				if (matches) {
+					if (invoke!==false) {
+						// copy matches onto props
+						Object.assign(attributes, { url, matches }, matches);
```

Comment:
It was already in use in the Link function (but think that might have been a recent change: in the last few months or so...)

```js
const Link = (props) => {
        return h('a', Object.assign({}, props, { onClick: handleLinkClick }));
};
```